idaes/models/properties/modular_properties/phase_equil/smooth_VLE.py

```diff
@@ -50,116 +50,288 @@ def phase_equil(b, phase_pair):
             )
         )
 
-    # Definition of equilibrium temperature for smooth VLE
+    # Definition of equilibrium temperature slack variables for smooth VLE
     t_units = b.params.get_metadata().default_units["temperature"]
     if v_only_comps == []:
         b.add_component(
-            "_t1" + suffix,
-            Var(
-                initialize=b.temperature.value,
-                doc="Intermediate temperature for calculating Teq",
-                units=t_units,
-            ),
+            "s" + suffix,
+            Var(b.params.phase_list,
+                initialize=0.0,
+                bounds=(0, None),
+                doc='Slack variable for equilibrium temperature',
+                units=t_units),
         )
-        _t1 = getattr(b, "_t1" + suffix)
+        s = getattr(b, "s" + suffix)
 
-        b.add_component(
-            "eps_1" + suffix,
-            Param(
-                default=0.01,
-                mutable=True,
-                doc="Smoothing parameter for Teq",
-                units=t_units,
-            ),
-        )
-        eps_1 = getattr(b, "eps_1" + suffix)
-
-        # PSE paper Eqn 13
-        def rule_t1(b):
-            return _t1 == smooth_max(
-                b.temperature, b.temperature_bubble[phase_pair], eps_1
-            )
+        # Equilibrium temperature
+        def rule_tbar(b):
+            return b.tbar[phase_pair] - b.temperature - \
+                s['Vap'] + s['Liq'] == 0
+        b.add_component("tbar_constraint" + suffix, Constraint(rule=rule_tbar))
 
-        b.add_component("_t1_constraint" + suffix, Constraint(rule=rule_t1))
     else:
-        _t1 = b.temperature
+        b.tbar[phase_pair] = b.temperature
 
-    if l_only_comps == []:
-        b.add_component(
-            "eps_2" + suffix,
-            Param(
-                default=0.0005,
-                mutable=True,
-                doc="Smoothing parameter for Teq",
-                units=t_units,
-            ),
-        )
-        eps_2 = getattr(b, "eps_2" + suffix)
+    b.add_component(
+        "eps" + suffix,
+        Param(default=1e-04,
+              mutable=True,
+              doc='Smoothing parameter for complementarities',
+              units=t_units),
+    )
+    eps = getattr(b, "eps" + suffix)
+    
+    b.add_component(
+        "gp" + suffix,
+        Var(b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc='Slack variable for cubic second derivative for phase p',
+            units=pyunits.dimensionless),
+    )
+    gp = getattr(b, "gp" + suffix)
+    
+    b.add_component(
+        "gn" + suffix,
+        Var(b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc='Slack variable for cubic second derivative for phase p',
+            units=pyunits.dimensionless),
+    )
+    gn = getattr(b, "gn" + suffix)
```

Comment:
FYI, you can replace blocks like this with:

```
    b.add_component("gn" + suffix, gn)
    gn = Var(b.params.phase_list,
            initialize=0.0,
            bounds=(0, None),
            doc='Slack variable for cubic second derivative for phase p',
            units=pyunits.dimensionless)
```

Reply:
Thanks, I have incorporated this